embedded-workshop-book/src/setup-stage.md

```diff
@@ -7,54 +7,66 @@ In `usb-2.rs`, you will find a short description of each register above the vari
+
+**2. Description of GET_DESCRIPTOR request**
 
 We can recognize a GET_DESCRIPTOR request by the following properties:
 - `bmRequestType` is **0b10000000**
 - `bRequest` is **6** (i.e. the GET_DESCRIPTOR Request Code, defined in table 9-4 in the USB spec)
 
+
+**3. Description of DEVICE descriptor requests**
 In this task, we only want to parse DEVICE descriptor requests. They have the following properties:
 
 - the descriptor type is **1** (i.e. DEVICE, defined in table 9-5 of the USB spec)
 - the descriptor index is **0**
 - the wIndex is **0** for our purposes
 - ❗️you need to fetch the descriptor type from the high byte of `wValue`, and the descriptor index from the the low byte of `wValue`
 
+Check section 9.4.3 of the [USB specification][usb_spec] for a very detailed description of the requests. All the constants we'll be using are also described in Tables 9-3, 9-4 and 9-5 of the same document.
+
+**4. Remember that you can define binary literals by prefixing them with `0b`.**
+
+**5. You can use bit shifts (`>>`) and casts (`as u8`) to get the high/low bytes of `wValue`.**
+
+**6. Return `Err` if properties aren't met.**
+
 
 You will also find this information in the `// TODO implement ...` comment in the `Request::parse()` function of `lib.rs` file.
  > NOTE: If you'd like to learn more, take a look at Section 9.4.3 Get Descriptor of the USB specification.
 
-To complete the task, proceed like this:
+See `advanced/common/usb/solution-get-descriptor-device.rs` for a solution.
 
-1. **Parse GET_DESCRIPTOR requests for DEVICE descriptors:**  
-Modify `Request::parse()` in `advanced/common/usb/src/lib.rs` to recognize a GET_DESCRIPTOR request of type DEVICE so that the `get_descriptor_device` test passes. Note that the parser already handles SET_ADDRESS requests.
+✅ Read incoming request information and pass it to the parser:
 
-    - remember the GET_DESCRIPTOR fields described at the start of this section
-    - remember that you can define binary literals by prefixing them with `0b`
-    - you can use bit shifts (`>>`) and casts (`as u8`) to get the high/low bytes of `wValue`
+modify `usb-2.rs` to read `USBD` registers and parse the SETUP data when an EP0SETUP event is received.
 
-See `advanced/common/usb/solution-get-descriptor-device.rs` for a solution.
+**Getting Started:**
 
-2. **Read incoming request information and pass it to the parser:**  
-modify `usb-2.rs` to read `USBD` registers and parse the SETUP data when an EPSETUP event is received.
     - for a mapping of register names to the `USBD` API, check the entry for `nrf52840_hal::target::usbd` in the documentation you've created using `cargo doc`
```

Comment:
I think you'll need to remove the indentation from this bullet list. Right now it's showing as a code block.